src/fsharp/TastPickle.fs

```diff
@@ -2121,23 +2121,31 @@ and u_ValData st =
         (u_option u_const) 
         (u_used_space1 u_xmldoc)
         st
-    { val_logical_name=x1
-      val_compiled_name=x1z
-      val_range=(match x1a with None -> range0 | Some(a,_) -> a)
-      val_other_range=(match x1a with None -> None | Some(_,b) -> Some(b,true))
-      val_type=x2
-      val_stamp=newStamp()
-      val_flags=ValFlags(x4)
-      val_defn = None
-      val_member_info=x8
-      val_attribs=x9
-      val_repr_info=x10
-      val_xmldoc= defaultArg x15 XmlDoc.Empty
-      val_xmldocsig=x12
-      val_access=x13
-      val_declaring_entity=x13b
-      val_const=x14
-    }
+    let res =
+        { val_logical_name=x1
+          val_range=(match x1a with None -> range0 | Some(a,_) -> a)
+          val_type=x2
+          val_stamp=newStamp()
+          val_flags=ValFlags(x4)
+          val_opt_data=None
+        }
+
+    res.val_opt_data <-
```

Comment:
Instead of the mutation you could set it directly here, right?

Reply:
Yes I can but maybe @dsyme  wanted to do this in a way that it's done in tast.fs

Reply:
was just an idea. I think it would cleanup the code here nicely